core/src/main/java/com/sun/mail/util/WriteTimeoutSocket.java

```diff
@@ -405,8 +424,9 @@ public synchronized void write(byte[] bs, int off, int len)
 		    sf = ses.schedule(timeoutTask,
 					timeout, TimeUnit.MILLISECONDS);
 	    } catch (RejectedExecutionException ex) {
-		// ignore it; Executor was shut down by another thread,
-		// the following write should fail with IOException
+            IOException ioex = new IOException("Exception writing in socket");
+            ioex.initCause(ex);
```

Comment:
Change "Exception writing in socket" to something like "Write aborted due to timeout not enforced."  

We could even query the socket to see if it was closed and provide that information in the message too.  Something to think about since that detail is lost because we no longer allow the attempted write.

Reply:
message fixed and added status of the socket

Reply:
A test should be added to show that a shutdown executor with an open socket throws the new exception with the chained rejected execution exception.